src/boomer.nim

```diff
@@ -8,14 +8,24 @@ import x11/xlib, x11/x, x11/xutil, x11/keysym
 import opengl, opengl/glx
 import la
 
+type Shader = tuple[path, content: string]
+
+proc readShader(file: string): Shader =
+  result.path = file
+  result.content = block:
+    when nimvm:
```

Comment:
What does `nimvm` exactly indicate? Is it for the hot reloading thingy?

Reply:
it tells if this was executed on compile time (in nim's vm) or at runtime, to dispatch correct procedure. slurp reads file relative to the source file where its invoked. readFile read relative to current working directory.

Reply:
this is added complexity for minor gain. i could just prefix the shader path with "src/", since `nimble build` will always be executed from the same directory.